codeforamerica/vita-min
# TBE 85 Bugcrowd: Add verification for emails for unsubscribe/subscribe

## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/TBE-85

## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

## What was done?
A bugrowd issue was brought up where any bad actor can pass in an email address to the unsubscribe page [en/unsubscribe_from_emails?email_address=] and can unsubsribe someone else. We added verification so that one can only unsubscribe an email address if they click the link from an email it is in, otherwise it the page will say that it is "invalid link"

## How to test?
Check on being a bad actor: Check to see on heroku using the following https://gyr-review-app-4663-534256ffcb66.herokuapp.com/en/unsubscribe_from_emails?email_address=<"email string"> and passing in a email address or any string. It should say invalid unsubscribtion link. 

You can also check that the origin flow and app work as before. This can be done by completing and intake for FYST or GYR on the heroku link, getting a welcome email, clicking the unsubscribe link and checking on the client in the hub to see that they are unsubscribed.
